app/services/openid_connect_service.rb

```diff
@@ -0,0 +1,12 @@
+class OpenidConnectService
+  def self.generate_authorization_code(user:, client_id:, nonce:)
+    identity = user.identities.create!(
+      service_provider: client_id,
+      session_uuid: SecureRandom.uuid,
+      last_authenticated_at: Time.current,
+      nonce: nonce
+    )
+
+    identity.session_uuid
+  end
```

Comment:
Would it make sense if we called it `generate_session_uuid` and assign that to the `code`?